consensus/service/src/timestamp_validator.rs

```diff
@@ -0,0 +1,236 @@
+// Copyright (c) 2023 The MobileCoin Foundation
+
+use displaydoc::Display;
+use mc_blockchain_types::Block;
+
+/// Provides logic for validating a timestamp used in consensus
+
+const MAX_TIMESTAMP_AGE: u64 = 30 * 1000; // 30 seconds
+
+/// Errors related to validating timestamps
+#[derive(Clone, Debug, Display, Eq, PartialEq)]
+pub enum Error {
+    /// The timestamp is in the future now: {0}, timestamp: {1}
+    InFuture(u64, u64),
+    /** The timestamp is not newer than the last bock. last_bock timestamp: {0},
+    timestamp: {1} */
+    NotNewerThanLastBlock(u64, u64),
+    /// The timestamp is too far in the past now: {0}, timestamp: {1}
```

Comment:
From, https://github.com/mobilecoinfoundation/mobilecoin/pull/3711#discussion_r1393486807
> Similarly, combining the timestamp with the word before it seems a bit weird to me

perhaps
"The current block time is too old, current time {0}, block time: {1}"
IDk the "current block" and "current time" is confusing, I think "now" feels a bit off in the error message.